numpyro/compat/infer.py

```diff
@@ -131,6 +132,8 @@ def step(self, *args, rng=None, **kwargs):
                                 'dicts of arrays.')
             else:
                 raise e
+        params = super(SVI, self).get_params(self.svi_state)
```

Comment:
I think we should `jit` this.

Reply:
Sure, will do that. Do you think this will have significant overhead?

Reply:
Yes, especially for models with many constrained supports.

Reply:
Yeah, that's a good point. I wish there was a way to signal the end of learning with Pyro's interface. Pyro's param store will do this lazily, but we'll be inefficiently doing this per `svi.step`. :(

Reply:
Can you explain for me how this is less efficient than Pyro? Even in Pyro constrained params must be transformed once per svi step.

Reply:
I meant this will have significant overhead when we didn't jit it. In NumPyro, params are already transformed to compute elbo. The transform here is not necessary unless we want to access param values from the param store. But I think this overhead is small because we already jit it.

Reply:
Actually, I think the only difference is that pyro will compute and cache this when it calls `pyro.param` whereas we will be doing this at least twice (once in inference and then again to put the parameters in the param store). But hopefully, with JIT, this shouldn't be as much of an issue.

Reply:
@neerajprad How about we resolve this by changing [this line](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/infer/svi.py#L205) to
```
def loss_fn(x):
     constrained_params = self.constrain_fn(x)
     return self.loss.loss(rng_seed, constrained_params, ...), constrained_params

(loss, constrained_params), grad = value_and_grad(loss_fn, has_aux=True)(params)
```
Basically, we use `has_aux=True` for constrained_params and move it to `svi_state`.

Reply:
That sounds reasonable. You mean, add another field called `constrained_params` to `SVIState`?